atest/regexp_prompt.robot

```diff
@@ -0,0 +1,38 @@
+*** Settings ***
+Default Tags   pybot   jybot
+Resource       resources/common.robot
+Test Teardown  Close All Connections
+
+*** Test Cases ***
+Read Until Prompt With Regexp Prompt
+    Open Connection  localhost  prompt=[$|#]  prompt_is_regexp=yes
+    Login  test  test
+    Write  pwd
+    ${output}=	Read Until Prompt
+    Should Contain	${output}  $
+    Write  su test_prompt
+    ${password}=	Read	delay=0.5s
+    Should Contain	${password}	Password:
+    Write  test
+    ${output1}=	Read Until Prompt
+    Should Contain	${output1}  \#
+    Write  exit
+    ${output2}=	Read Until Prompt
+    Should Contain	${output2}  $
```

Comment:
I'd propose simpler test with configuring PS1 instead of adding another test user. Something like:
```robotframework
Read Until Prompt With Regexp Prompt
    Open Connection  localhost  prompt=[$|#]  prompt_is_regexp=yes
    Login  test  test
    Write  pwd
    ${output}=	Read Until Prompt
    Should Contain	${output}  $
    Write  export PS1='\u@\h \W \# '
    ${output1}=	Read Until Prompt
    Should Contain	${output1}  \#
    [Teardown]  Close connection
```
With this approach there is no need for new test_prompt user.

Reply:
I wouldn't go changing environment settings during test execution. The change could be reverted in teardown section, but in case of a connection error or system crash the environment would not reverted to the previous state and the rest of the test cases would fail.

As I see it, the safest bet is to have another test user.

Reply:
No, `export PS1` affects only current session, no need for reverting in teardown and won't impact rest of the test cases. Try it.

Reply:
My mistake. Yes, you're right.